client/components/Recipe.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+
+export default function Recipe() {
+  const { id } = useParams()
+  const recipes = useSelector((state) => state.recipes)
+  const recipe = recipes[id]
+
+  const label = recipe.recipe.label
+  const largeImage = recipe.recipe.images.LARGE?.url
+  const regularImage = recipe.recipe.images.REGULAR.url
+  const ingredientsArr = recipe.recipe.ingredients
+
+  const dietaryArr = recipe.recipe.healthLabels
+  const selectiveDietary = dietaryArr.filter((word) => {
```

Comment:
since this is dependent on `dietaryArr` you should either have this within a `useEffect` or a `useMemo` and have `dietaryArr` in the dependancy array

Reply:
Because all of this stuff isnt wrapped in a useMemo or useEffect, All of these will run again on each re-render. The stuff above doesnt matter to much because they are just getting values from your recipe, but this one is doing some logic which takes time and is something you dont want to run on every re-render.

To prevent this you wrap it either a `useMemo` or a `useEffect` and have `dietaryArr` in the dependancy array so that it will only change if `dietaryArr` changes. Since I imagine you guys dont know `useMemo` yet (which is what I would do by the way), I would then just make a state for selectiveDietary and have the useEffect set it

Reply:
Thanks Richard. This is a very useful tip. I will approach this at some stage